fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -1958,4 +2009,236 @@ TEST_CASE("/backup/continuous") {
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 199, 200));
+
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 250, 260));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 250, 310));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 250, 405));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 400, 401));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 401, 450));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 300, 399));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 300, 400));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 300, 401));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 300, 350));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 350, 370));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 350, 400));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 10, 400));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 100, 400));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 200, 400));
+	ASSERT(!BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 299, 400));
+	ASSERT(BackupContainerFileSystemImpl::hasContinuousLogsForSnapshot(files, 399, 400));
+
+	return Void();
+}
+
+void printFileList(BackupFileList& backupFileList) {
+	printf("\nRangeFiles count:%d", backupFileList.ranges.size());
+	for (auto r : backupFileList.ranges)
+		printf("\n%s", r.toString().c_str());
+
+	printf("\nLogFiles count:%d", backupFileList.logs.size());
+	for (auto l : backupFileList.logs)
+		printf("\n%s", l.toString().c_str());
+
+	printf("\nSnapshotFiles count:%d", backupFileList.snapshots.size());
+	for (auto s : backupFileList.snapshots)
+		printf("\n%ld, %ld, %s, %ld", s.beginVersion, s.endVersion, s.fileName, s.totalSize);
+}
+
+// Intentionally missing some log range files and checking if the snapshot can be restored.
+ACTOR Future<Void> testBackupContainerWithMissingLogRanges(std::string url, Optional<std::string> proxy) {
+	state FlowLock lock(100e6);
+	printf("BackupContainerTest URL %s\n", url.c_str());
+
+	state Reference<IBackupContainer> c = IBackupContainer::openContainer(url, proxy, {});
+	// Make sure container doesn't exist, then create it.
+	try {
+		wait(c->deleteContainer());
+	} catch (Error& e) {
+		if (e.code() != error_code_backup_invalid_url && e.code() != error_code_backup_does_not_exist)
+			throw;
+	}
+	wait(c->create());
+
+	state Key begin = ""_sr;
+	state Key end = ""_sr;
+	state int blockSize = 3 * sizeof(uint32_t) + begin.size() + end.size() + 8;
+	state std::vector<Future<Void>> writes;
+	state std::pair<Key, Key> beginEndKeys = std::make_pair(begin, end);
```

Comment:
both keys are empty. i guess you want to have files with different ranges. 

`getRestoreSet()` calls `readKeyspaceSnapshot()`, which checks if the snapshot covers the ranges to be restored. So it might be a good idea to test the range file coverage as well.

Reply:
I was confused why rangeFile creation is not taking keys and only snapshot creation is taking keys. So, I ignored.
I will add random keys just incase it is useful.